### PR TITLE
Do not use oauth client without token

### PR DESCRIPTION
**Pull request by qwerty287**
Closes https://github.com/woodpecker-ci/woodpecker/issues/1370

### CODE REVIEW

File: server/forge/github/github.go [hunk 1]
```diff
@@ -394,16 +394,29 @@ func (c *client) newConfig(req *http.Request) *oauth2.Config {
...
-	)
-	tc := oauth2.NewClient(ctx, ts)
-	if c.SkipVerify {
-		tc.Transport.(*oauth2.Transport).Base = &http.Transport{
-			Proxy: http.ProxyFromEnvironment,
-			TLSClientConfig: &tls.Config{
-				InsecureSkipVerify: true,
-			},
+	var tc *http.Client
+	if token != "" {
```
Comment (anbraten) [inline server/forge/github/github.go:398]: Why could the token be empty?

Comment (anbraten) [reply server/forge/github/github.go:398]: I think that could be an issue if the repo is private and the woodpecker project is public. Could we pass the token of the project / repo instead?

File: server/forge/common/utils.go [hunk 2]
```diff
@@ -37,3 +43,20 @@ func ExtractHostFromCloneURL(cloneURL string) (string, error) {
...
+	if u != nil {
+		return u.Token
+	}
+
+	_store, ok := store.TryFromContext(ctx)
+	if !ok {
+		log.Error().Msg("could not get store from context")
+		return ""
+	}
+	user, err := _store.GetUser(r.UserID)
```
Comment (6543) [inline server/forge/common/utils.go:57]: hmm we might als add a nil check for r

Comment (6543) [reply server/forge/common/utils.go:61]: I added a simple check that in best case will never be exec, in worst case we have a unique error msg we can trace down

**Pull request merged**